### ISSUE TITLE
Ignore line-no from dbt mismatch linter's violation line-no 

### ISSUE THREAD

**Issue by raphael-riel** [sqlfluff/sqlfluff#1008]
Code:
![image](https://user-images.githubusercontent.com/474697/116128819-5cac4b80-a697-11eb-9746-438b531c1b2a.png)
Violation L029 is at line 32. Also it's ignored at the line-level.

![image](https://user-images.githubusercontent.com/474697/116128911-76e62980-a697-11eb-9ec1-fd5316a019fe.png)

Ignore mask's line-no is off. 
![image](https://user-images.githubusercontent.com/474697/116129227-dc3a1a80-a697-11eb-8826-233eb1e19487.png)

Looks like it's coming from `EnrichedFilePositionMarker`
![image](https://user-images.githubusercontent.com/474697/116130916-c4fc2c80-a699-11eb-8026-084355848a56.png)

I confirm the compiled dbt's file is NOT loaded in any way.

## Expected Behaviour
I was expecting my `noqa` directive to make this violation ignored.

## Observed Behaviour
Line from IgnoreMask and violation doesn't match, even if they are in the same file, on the same line.

## Version
Include the output of `sqlfluff --version` along with your Python version

```
sqlfluff, version 0.5.2
Python 3.8.7
dbt 0.19.1
dialect: snowflake
```

## Configuration
```
[sqlfluff]
templater = dbt
dialect = snowflake

# TODO: L039 -> We have to decide on how things are aligned in SELECT clause.
# L009 -> dbt templating does not keep trailing new lines (L009)
exclude_rules = L039,L009

[sqlfluff:rules:L016]
max_line_length=120
```

Ignore:
```
target/
dbt_modules/
macros/

```

**Comment by raphael-riel** [comment]
I'm currently trying to figure out what's going on at https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/core/parser/segments/base.py#L310
That logic would be my first hypothesis for the problem.

**Comment by raphael-riel** [comment]
Ok, found the culprit: Config node written in jinja for specific values in a model.


## Fail
![2021-04-26 at 2 59 PM](https://user-images.githubusercontent.com/474697/116136796-cbda6d80-a6a0-11eb-8ea6-0b7c047f9a0b.png)

## Success
![image](https://user-images.githubusercontent.com/474697/116136824-d4cb3f00-a6a0-11eb-9b22-1dd2a6e25cdb.png)

## SQL to Reproduce

```sql
{{ config(
    cluster_by=['some_column'],
    alias='my_alias'
) }}

select
    uuid,
    something as value  -- noqa: L029

from some_table

```

**Comment by raphael-riel** [comment]
## Update/workaround
Putting the jinja-config-call on a single line seems to be a valid workaround:
![image](https://user-images.githubusercontent.com/474697/116137172-37243f80-a6a1-11eb-9ff2-cbcf637a5842.png)


## Fails
```sql
{{ config(
    cluster_by=['some_column'],
    alias='my_alias'
) }}

select
    uuid,
    something as value  -- noqa: L029

from some_table
```

## Works
```sql
{{ config(cluster_by=['some_column'], alias='my_alias') }}

select
    uuid,
    something as value  -- noqa: L029

from some_table
```

**Comment by raphael-riel** [comment]
The same seems to apply to using macro's output in a CTE.
![image](https://user-images.githubusercontent.com/474697/116139670-45278f80-a6a4-11eb-8a0c-e28b1f983c4b.png)
![image](https://user-images.githubusercontent.com/474697/116139718-51135180-a6a4-11eb-80e5-36e961a57183.png)

Using macro's content directly fixes the issue
![image](https://user-images.githubusercontent.com/474697/116139831-7607c480-a6a4-11eb-9a3f-1387fedcaa45.png)

**Comment by raphael-riel** [comment]
Looks like there's either:

A) Something wrong with my setup/config making jinja's output fail `noqa` directives
B) Parser's miscalculating jinja's render outputs.

Can someone test out what I presented in https://github.com/sqlfluff/sqlfluff/issues/1008#issuecomment-827077797 with their dbt-setup?

Thanks!

**Comment by NiallRees** [comment]
Hey @raphael-riel I'm not able to reproduce this using the latest master commit. It looks like you've dived into the code though which is great to see! If you are still facing this error, would you be able to put together a test case which demonstrates it? That would be super helpful 🙏

**Comment by raphael-riel** [comment]
@NiallRees Thanks for trying to repro this!

Here's everything to repro in 0.5.2. 
https://github.com/raphael-riel/sqlfluff-issue-1008/

0.5.2 because 0.5.3 did fix the issue!!!

You can see reporting the error here: https://github.com/raphael-riel/sqlfluff-issue-1008/runs/2517959437?check_suite_focus=true
Yet the ignore is properly set here: https://github.com/raphael-riel/sqlfluff-issue-1008/blob/main/dbt/test/models/test.sql#L8

I quickly scanned the release diff (https://github.com/sqlfluff/sqlfluff/compare/0.5.2...0.5.3) and found that https://github.com/sqlfluff/sqlfluff/compare/0.5.2...0.5.3#diff-fe70b9c424984a64d62d1fd3dc9c3b3877ea0d2de3331b2f27fd56863bd57520R504 have fixed my issue. PR https://github.com/sqlfluff/sqlfluff/pull/978

Seems like that I opened this issue right after the PR+Issues were merged and closed 😄 

Thanks for this great project! Now my humongous dbt's collection of SQL files can be kept under control! Thanks Team!
